app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/RegisterWithEoriAndIdController.scala

```diff
@@ -132,37 +132,37 @@ class RegisterWithEoriAndIdController @Inject() (
         case Some("DEFERRED") =>
           val formattedDate =
             dateTimeFormat.print(resp.responseCommon.processingDate)
-          Future.successful(Redirect(RegisterWithEoriAndIdController.pending(formattedDate)))
+          Future.successful(Redirect(RegisterWithEoriAndIdController.pending(service, formattedDate)))
         case Some("FAIL") =>
           val formattedDate =
             dateTimeFormat.print(resp.responseCommon.processingDate)
-          Future.successful(Redirect(RegisterWithEoriAndIdController.fail(formattedDate)))
+          Future.successful(Redirect(RegisterWithEoriAndIdController.fail(service, formattedDate)))
         case None =>
           val statusText = resp.responseCommon.statusText
-          handleErrorCodes(statusText)
+          handleErrorCodes(service, statusText)
         case _ =>
           CdsLogger.error("Unknown RegistrationDetailsOutCome ")
           throw new IllegalStateException("Unknown RegistrationDetailsOutCome")
       }
     }
 
-  def processing: Action[AnyContent] = ggAuthorisedUserWithEnrolmentsAction {
+  def processing(service: Service): Action[AnyContent] = ggAuthorisedUserWithEnrolmentsAction {
```

Comment:
For this method and all below you're not passing service to the view.

This will be a part of the story where we will be changing content there?

Reply:
For now, the view doesn't need the service to be passed.  The header and layout components work out the service based on the url.  Other part's of the view could do this as well (e.g. to use it to pass to the "submit" method for the next page) but I think it's safer to pass in the service in these cases - just in case the url doesn't contain "atar" for some reason.

As you said, when we update the content on these "confirmation" pages (other than the title) we can update to pass in the service.